test/pycheck/temporary_table_test.py

```diff
@@ -0,0 +1,22 @@
+from .utils import Cursor, PG_MAJOR_VERSION, eprint
+
+
+def test_temporary_table_set_access_method(cur: Cursor):
+    if PG_MAJOR_VERSION >= 15:
+        cur.sql("CREATE TEMP TABLE t(a int)")
+        cur.sql("ALTER TABLE t ADD COLUMN b int")
+
+        try:
```

Comment:
Instead of the `try` this should use `with pytest.raises(...):` that way the test will fail if the intended exception is not thrown.